internal/dag/secret.go

```diff
@@ -0,0 +1,82 @@
+// Copyright © 2019 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dag
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// validateSecret returns true if the secret is interesting and well formed.
+func validateSecret(secret *v1.Secret) (bool, error) {
```

Comment:
nit: Should the name of this method be more descriptive? Like `validSecret` or something so the caller understands that `true` is valid.

Reply:
Will do. Sadly `validSecret` is already taken by the callback we pass to `lookupSecret`.